doc: nrf5340audio: add docs for new apps (#37)
alexsven/sdk-nrf: applications/nrf5340_audio/broadcast_source/README.rst

.. _nrf53_audio_broadcast_source_app:

nRF5340 Audio: Broadcast source
###############################

.. contents::
   :local:
   :depth: 2

The nRF5340 Audio broadcast source application implements the :ref:`BIS gateway mode <nrf53_audio_app_overview>`.

In this mode, transmitting broadcast audio happens using Broadcast Isochronous Stream (BIS) and Broadcast Isochronous Group (BIG).
Play and pause are emulated by disabling and enabling stream, respectively.

The following limitations apply to this application:

* One BIG with two BIS streams.
* Audio input: USB or I2S (Line in or using Pulse Density Modulation).
* Configuration: 16 bit, several bit rates ranging from 32 kbps to 124 kbps.

.. _nrf53_audio_broadcast_source_app_requirements:

Requirements
************

The application shares the :ref:`requirements common to all nRF5340 Audio application <nrf53_audio_app_requirements>`.

.. _nrf53_audio_broadcast_source_app_ui:

User interface
**************

This application uses specific mapping for the following user interface elements:

> * **BTN 4** - Pressed on the gateway during playback: Sends a test tone generated on the device. Use this tone to check the synchronization of headsets.

Reviewer: Include: ``**PLAY/PAUSE** - Pressed on the broadcast source during playback: Starts/stops the stream, depending on the current status of the stream``, or something similar

Author: Do we need this? Isn't PLAY/PAUSE identical to all apps?

+-------------------+---------------------------------+-----------------------------------------------------------------+

.. _nrf53_audio_broadcast_source_app_testing:

Testing
*******

<VERIFY>

.. _nrf53_audio_broadcast_source_app_testing_steps_bis:

Testing the BIS mode
====================

Testing the BIS mode is identical to Testing the default CIS mode, except for the following differences:

* You can play the audio stream with different audio settings on the receivers.
  For example, you can decrease or increase the volume separately for each receiver during playback.
* When pressing the **PLAY/PAUSE** button on a headset, the streaming state only changes for that given headset.
* Pressing the **PLAY/PAUSE** button on the gateway will respectively start or stop the stream for all headsets listening in.
* Pressing the **BTN 4** button on a headset will change the active audio stream.
  The default configuration of the BIS mode supports two audio streams (left and right).
* Pressing the **BTN 5** button on a headset will change the gateway source for the audio stream.

>   If a second gateway is not present, the headset will not play audio.

Reviewer: Should this be here? For the gateway side the only things you can do are already highlighted in the user interface section

Author: Let me schedule a meeting about the Testing sections.

.. _nrf53_audio_broadcast_source_app:

nRF5340 Audio: Broadcast source
###############################

.. contents::
   :local:
   :depth: 2

The nRF5340 Audio broadcast source application implements the :ref:`BIS gateway mode <nrf53_audio_app_overview>`.

In this mode, transmitting broadcast audio happens using Broadcast Isochronous Stream (BIS) and Broadcast Isochronous Group (BIG).

> Play and pause are emulated by disabling and enabling stream, respectively.

Reviewer 2: ```suggestion
Play and pause are emulated by enabling and disabling the stream, respectively.
```

> Play and pause are emulated by disabling and enabling stream, respectively.

Reviewer: ```suggestion
Play and pause are emulated by enabling and disabling stream, respectively.
```

The following limitations apply to this application:

* One BIG with two BIS streams.
* Audio input: USB or I2S (Line in or using Pulse Density Modulation).
* Configuration: 16 bit, several bit rates ranging from 32 kbps to 124 kbps.

.. _nrf53_audio_broadcast_source_app_requirements:

Requirements
************

The application shares the :ref:`requirements common to all nRF5340 Audio application <nrf53_audio_app_requirements>`.

.. _nrf53_audio_broadcast_source_app_ui:

User interface
**************

Most of the user interface mappings are common across all nRF5340 Audio applications.
See the :ref:`nrf53_audio_app_ui` page for detailed overview.

This application uses specific mapping for the following user interface elements:

* Pressed on the broadcast source device during playback:

  * **PLAY/PAUSE** - Starts or pauses the playback of the stream.

>   * **BTN 4** -  Sends a test tone generated on the device.

Reviewer 2: ```suggestion
  * **BTN 4** -  Toggles between the normal audio stream and different test tones generated on the device.
```

    Use this tone to check the synchronization of headsets.

> * **LED1** - Blinking blue on the broadcast source device: Device has started broadcasting audio.

Reviewer: ```suggestion
* **LED1** - Blinking blue: Device has started broadcasting audio.
```

* **RGB1** - Solid green - The device is programmed as the gateway.

.. _nrf53_audio_broadcast_source_app_configuration:

Configuration
*************

See :ref:`nrf53_audio_app_configuration` and :ref:`nrf53_audio_app_fota` for configuration options common to all nRF5340 Audio applications.

For information about how to configure applications in the |NCS|, see :ref:`modifying`.

.. _nrf53_audio_broadcast_source_app_building:

Building and running
********************

This application can be found under :file:`applications/nrf5340_audio/broadcast_source` in the nRF Connect SDK folder structure.

The nRF5340 Audio DK comes preprogrammed with basic firmware that indicates if the kit is functional.
See :ref:`nrf53_audio_app_dk_testing_out_of_the_box` for more information.

To build the application, see :ref:`nrf53_audio_app_building`.

After programming, the broadcast source automatically starts broadcasting the default 48-kHz audio stream.

.. _nrf53_audio_broadcast_source_app_testing:

> Testing

Reviewer 2: Ref. earlier comment.
